go.mod

```diff
@@ -3,35 +3,37 @@ module github.com/prometheus/pushgateway
 require (
 	github.com/go-kit/log v0.2.1
 	github.com/golang/protobuf v1.5.2
-	github.com/matttproud/golang_protobuf_extensions v1.0.1
-	github.com/prometheus/client_golang v1.12.2
-	github.com/prometheus/client_model v0.2.0
-	github.com/prometheus/common v0.34.0
-	github.com/prometheus/exporter-toolkit v0.7.1
+	github.com/matttproud/golang_protobuf_extensions v1.0.4
+	github.com/prometheus/client_golang v1.14.0
+	github.com/prometheus/client_model v0.3.0
+	github.com/prometheus/common v0.37.0
+	github.com/prometheus/exporter-toolkit v0.8.1
 	github.com/shurcooL/vfsgen v0.0.0-20200824052919-0d455de96546
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
 )
 
 require (
 	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 // indirect
-	github.com/alecthomas/units v0.0.0-20190924025748-f65c72e2690d // indirect
+	github.com/alecthomas/units v0.0.0-20211218093645-b94a6e3cc137 // indirect
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/cespare/xxhash/v2 v2.1.2 // indirect
+	github.com/coreos/go-systemd/v22 v22.4.0 // indirect
 	github.com/go-logfmt/logfmt v0.5.1 // indirect
 	github.com/jpillora/backoff v1.0.0 // indirect
 	github.com/julienschmidt/httprouter v1.3.0 // indirect
 	github.com/mwitkow/go-conntrack v0.0.0-20190716064945-2f068394615f // indirect
-	github.com/pkg/errors v0.9.1 // indirect
-	github.com/prometheus/procfs v0.7.3 // indirect
+	github.com/prometheus/procfs v0.8.0 // indirect
 	github.com/shurcooL/httpfs v0.0.0-20190707220628-8d4bc4ba7749 // indirect
-	golang.org/x/crypto v0.0.0-20210616213533-5ff15b29337e // indirect
-	golang.org/x/net v0.0.0-20220225172249-27dd8689420f // indirect
-	golang.org/x/oauth2 v0.0.0-20220223155221-ee480838109b // indirect
-	golang.org/x/sys v0.0.0-20220114195835-da31bd327af9 // indirect
+	golang.org/x/crypto v0.0.0-20221012134737-56aed061732a // indirect
+	golang.org/x/net v0.0.0-20220909164309-bea034e7d591 // indirect
+	golang.org/x/oauth2 v0.0.0-20220909003341-f21342109be1 // indirect
+	golang.org/x/sync v0.1.0 // indirect
+	golang.org/x/sys v0.0.0-20220728004956-3c1f35247d10 // indirect
 	golang.org/x/text v0.3.7 // indirect
-	google.golang.org/appengine v1.6.6 // indirect
-	google.golang.org/protobuf v1.26.0 // indirect
+	google.golang.org/appengine v1.6.7 // indirect
+	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
-go 1.17
+go 1.19
```

Comment:
Is there a reason to update this to 1.19? We typically try and support the last two releases.

Reply:
Isn't that only a guideline for libraries but not for stand alone programs like the PGW?

Furthermore, setting this in `go.mod` doesn't prevent you from compiling with an older Go version. But it does change some other behavior. If I leave this at `go 1.17`, I get CI errors because `go mod tidy` has a different result than what I do locally. Maybe the CI actually uses Go 1.17 with the old directive for `go mod tidy`?

I don't understand the issue fully, but leaving this at `go 1.17` definitely creates an issue. If we can solve it in a different way, I'm all up for it.

Reply:
Update: I think I found a solution. Let's see what the CI says.

Reply:
Yea, there are some compat warnings with `go mod tidy`. I usually solve it by explicitly updating the thing it complains about.